Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1430,13 +1425,28 @@ void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizes()
     Vector<GridItemWithSpan> itemsSortedByIncreasingSpan;
     Vector<GridItemWithSpan> itemsCrossingFlexibleTracks;
     WeakHashSet<RenderBox> itemsSet;
+    WeakHashSet<RenderGrid> subgridsWithEdgesInTrack;
 
     if (m_grid.hasGridItems()) {
         for (auto trackIndex : m_contentSizedTracksIndex) {
+            auto addAccumulatedSubgrirdMarginBorderPaddingToTrackSize = [&] {
+                const auto oppositeAxis = m_direction == GridTrackSizingDirection::ForColumns ? GridTrackSizingDirection::ForRows : GridTrackSizingDirection::ForColumns;
+                Vector<LayoutUnit> accumulatedSubgridMarginBorderPaddingsInTrack(m_renderGrid->numTracks(oppositeAxis));
+
+                for (const auto& subgrid : subgridsWithEdgesInTrack) {
```

Comment:
I'm not quite sure I follow the logic here.

`subgridsWithEdgesInTrack` is the set all of sub grids that have a start/end track that is also this track. Two entries in that set might be a parent-child relationship (for a nested sub grid), or siblings.

We're taking the sum of the paddings for all sub grids that start in the same track in the opposite axis, and then take the max of those.

I don't understand why the opposite axis (and the sub grids start line in it) is the boundary at which we sum vs max.

Reply:
Is the assumption that if two entries share the same start line in the opposite axis, then they must be a parent-child/nested-subgrid relationship?

I don't think that's guaranteed, you can set forced grid placement on items to make multiple items share the same grid cell.

Reply:
Yeah I was attempting to use the opposite axis index as a way of determining how subgrids were nested, but that was not quite correct as your example shows. 


I went back and slightly modified the approach. Instead of creating a list of subgrids we can just maintain a variable that accumulates as we nest into subgrids. It seems like the current recursive nature of the calls to `accumulateIntrinsicSizesForTrack` correctly captures the nested subgrid structure.

Let me know what you think!